tasks/swarm_init.rb

```diff
@@ -6,16 +6,16 @@
 require 'puppet'
 
 def swarm_init(advertise_addr, autolock, cert_expiry, dispatcher_heartbeat, external_ca, force_new_cluster, listen_addr, max_snapshots, snapshot_interval)
-  cmd_string = 'docker swarm init'
-  cmd_string += " --advertise-addr=#{advertise_addr}" unless advertise_addr.nil?
-  cmd_string += ' --autolock' unless autolock.nil?
-  cmd_string += ' --cert-expiry' unless cert_expiry.nil?
-  cmd_string += " --dispatcher-heartbeat=#{dispatcher_heartbeat}" unless dispatcher_heartbeat.nil?
-  cmd_string += " --external-ca=#{external_ca}" unless external_ca.nil?
-  cmd_string += ' --force-new-cluster' unless force_new_cluster.nil?
-  cmd_string += " --listen-addr=#{listen_addr}" unless listen_addr.nil?
-  cmd_string += " --max-snapshots=#{max_snapshots}" unless max_snapshots.nil?
-  cmd_string += " --snapshot-interval=#{snapshot_interval}" unless snapshot_interval.nil?
+  cmd_string = ['docker', 'swarm', 'init']
+  cmd_string += [" --advertise-addr=#{advertise_addr}"] unless advertise_addr.nil?
+  cmd_string += [' --autolock'] unless autolock.nil?
+  cmd_string += [' --cert-expiry'] unless cert_expiry.nil?
+  cmd_string += [" --dispatcher-heartbeat=#{dispatcher_heartbeat}"] unless dispatcher_heartbeat.nil?
+  cmd_string += [" --external-ca=#{external_ca}"] unless external_ca.nil?
+  cmd_string += [' --force-new-cluster'] unless force_new_cluster.nil?
+  cmd_string += [" --listen-addr=#{listen_addr}"] unless listen_addr.nil?
+  cmd_string += [" --max-snapshots=#{max_snapshots}"] unless max_snapshots.nil?
+  cmd_string += [" --snapshot-interval=#{snapshot_interval}"] unless snapshot_interval.nil?
```

Comment:
We should try to avoid using += here.

I believe [Array#concat](https://ruby-doc.org/core-2.1.1/Array.html#method-i-concat) would be preferable in this instance.

Also lets rename the variable to cmd rather than cmd_string.

Reply:
See above.

Reply:
In this case you have changed from concatenating a string to an array so we should generally lean towards the more idiomatic approach.